compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -288,43 +295,32 @@ pub(crate) fn to_llvm_features<'a>(sess: &Session, s: &'a str) -> Option<LLVMFea
             | "vector-enhancements-3"
             | "vector-packed-decimal-enhancement-3",
         ) if get_version().0 < 20 => None,
-        // Enable the evex512 target feature if an avx512 target feature is enabled.
-        ("x86", s) if s.starts_with("avx512") => Some(LLVMFeature::with_dependencies(
-            s,
-            smallvec![TargetFeatureFoldStrength::EnableOnly("evex512")],
-        )),
-        // Support for `wide-arithmetic` will first land in LLVM 20 as part of
-        // llvm/llvm-project#111598
-        ("wasm32" | "wasm64", "wide-arithmetic") if get_version() < (20, 0, 0) => None,
-        ("sparc", "leoncasa") => Some(LLVMFeature::new("hasleoncasa")),
+        ("wasm32" | "wasm64", "wide-arithmetic") if get_version().0 < 20 => None,
+        (
+            "x86",
+            "amx-avx512" | "amx-fp8" | "amx-movrs" | "amx-tf32" | "amx-transpose" | "movrs",
+        ) if get_version().0 < 20 => None,
+        /*
+            Perform relatively complex feature conversion.
+        */
```

Comment:
This doesn't make sense to me because "simple" and "complex" are purely subjective here, and this redistributes the match arms for features for a single architecture across the match instead of putting them together. This is not a coherent organization strategy that makes sense with how patches usually land. Typically, architecturally focused patches affect one architecture at a time, so they should all be closer together.